### ISSUE TITLE
Valgrind leak tests for apache check root process only

### ISSUE THREAD

**Issue by yeputons** [apache/incubator-pagespeed-mod#1364]
We have a script `check_for_leaks` (not in open source right now) which runs apache under valgrind and reports leaks. Apache spawns children processes to process requests and it looks like valgrind does not trace these.

How to reproduce:
1. Add these lines in `system_caches.cc`:

```
// in the beginning of SystemCaches::RootInit()
LOG(ERROR) << "Leaking from RootInit: " << new int[1000 * 1000]; 
// in the beginning of SystemCaches::ChildInit()
LOG(ERROR) << "Leaking from ChildInit: " << new int[1000 * 1000]; 
```
1. Add `exit` to the very beginning of `automatic/system_test.sh`
2. Run `make apache_debug_restart`
3. Run `scripts/check_for_leaks --no_unit_tests localhost:8080` and look for the message after running apache with file cache. It will say that 4MB are leaked (one of our blocks), but logs say that several children have also leaked memory.

**Comment by yeputons** [comment]
What's even more interesting is that the scripts prints that `--trace-children=yes` option is passed to valgrind, so maybe it's a matter of filtering its output.

**Comment by oschaaf** [comment]
@yeputons I can't see the specifics off course, but perhaps this is relevant?:

http://stackoverflow.com/questions/30416082/preserving-the-content-in-valgrinds-log-files-when-using-trace-children-and-e

**Comment by yeputons** [comment]
I finally got a chance to look into that. False alarm, I'm sorry.

The problem was with my interpretation of logs: I looked in the end of log only, while Valgrind prints several "LEAK SUMMARY" blocks in different places of log, possibly one per process. If I simply grep the log for "definitely lost", I'm able to see my special leaks, and that's exactly what our script does to ensure there are no leaks.
